Sources/MapboxMaps/Foundation/DictionaryCoding/DictionaryEncoder.swift

```diff
@@ -164,19 +202,73 @@ private extension DictionaryEncoder {
         }
 
         func encode<T: Encodable>(_ value: T, forKey key: Key) throws {
-            if DictionaryEncoder.isSupportedType(T.self) {
-                storage[key.stringValue] = .value(value)
-            } else {
-                let result = try Encoder(codingPath: codingPath.appending(key: key), userInfo: userInfo).encode(value)
-                storage[key.stringValue] = .value(result)
-            }
+            let result = try DictionaryEncoder.encode(
+                value,
+                codingPath: codingPath.appending(key: key),
+                userInfo: userInfo,
+                shouldEncodeNilValues: shouldEncodeNilValues)
+            storage[key.stringValue] = .value(result)
         }
 
         func encodeIfPresent<T: Encodable>(_ value: T?, forKey key: Key) throws {
-            let shouldEncodeNilValue = userInfo[.shouldEncodeNilValues] as? Bool ?? false
-            if value != nil || shouldEncodeNilValue {
-                try encode(value, forKey: key)
-            }
+            guard value != nil || shouldEncodeNilValues else { return }
+            try encode(value, forKey: key)
+        }
+
+        func encodeIfPresent(_ value: Bool?, forKey key: Key) throws {
+            try encodeIfPresent(NilEncodable(value), forKey: key)
```

Comment:
What does wrapping the value into `NilEncodable` do here? For me it seems that it is the same as passing `value` directly, if the value is nil - nothing will be encoded, non-nil - it will be encoded.

Reply:
because these are overloading methods with same signature, so I wrap them in a struct so it will trigger `func encodeIfPresent<T: Encodable>(_ value: T, forKey key: Key)` instead, otherwise I would have to do the check for `shouldEncodeNilValues` in all the methods of `encodeIfPresent`
Another approach would be having a private method
```
private func encodeNilIfNeeded<T: Encodable>(_ value: T?, forKey key: Key) throws {
  guard value != nil || shouldEncodeNilValues else { return }
  try encode(value, forKey: key)
}
```
then have all the `encodeIfPresent` to point to this one, what do you think?